### PR TITLE
feat(Text): support multiline and fixed width ellipsis

### PR DESCRIPTION
**Pull request by IsaevAlexandr**

### CODE REVIEW

File: src/components/Text/text/text.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ export interface TextBaseProps {
      * - text-overflow: ellipsis;
      */
     ellipsis?: boolean;
+    /**
+     * The number of whole lines of text after which the content will be cut off.
+     *
+     * !Note: supports only modern browsers
+     * https://caniuse.com/?search=display%3A%20-webkit-box%3B
+     */
+    ellipsisLines?: 2 | 3 | 4 | 5;
```
Comment (amje) [inline src/components/Text/text/text.ts:80]: What if I want more than 5? Maybe let this to be set via JS, in style attribute?

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Text/Text.tsx [hunk 2]
```diff
@@ -57,19 +61,33 @@ export const Text = React.forwardRef(
...
+
         return (
             <Tag
                 ref={ref}
                 className={text(
-                    {variant, ellipsis, whiteSpace, wordBreak},
+                    {
+                        variant,
+                        ellipsis: Boolean(ellipsis),
+                        ellipsisLines,
```
Comment (amje) [inline src/components/Text/Text.tsx:84]: Maybe `ellipsisLines`  is better to be the `ellipsis` prop? Boolean is for normal ellipsis, Number is for line-clamp. Why we need this shortcut: `ellipsis: 200` === `style: {maxWidth: 200}`? It seems weird for me

Comment (IsaevAlexandr) [reply src/components/Text/Text.tsx]: Very often it is necessary to limit the length of the text to no more than a fixed value. Every time add a container for this - redundant

File: src/components/Text/text/text.ts (hunk 1 above)
Comment (IsaevAlexandr) [reply src/components/Text/text/text.ts]: added limit by 10 in text utility and unlimited lines count in Text compoennt

File: src/components/Text/Text.tsx (hunk 2 above)
Comment (amje) [reply src/components/Text/Text.tsx]: We can use `style: {maxWidth: <value>}`. Why put it into prop `ellipsis`, there is no relation to it

File: src/components/Text/text/text.scss [hunk 3]
```diff
@@ -68,8 +70,28 @@ $block: '.#{variables.$ns}text';
         @include mixins.text-subheader-3();
     }
     &_ellipsis {
+        display: inline-block;
         @include mixins.overflow-ellipsis();
     }
+
+    // ellipsis lines
+    &_el {
```
Comment (amje) [inline src/components/Text/text/text.scss:78]: Let's name it `_ellipsis-lines`

File: src/components/Text/text/text.scss [hunk 4]
```diff
@@ -68,8 +70,28 @@ $block: '.#{variables.$ns}text';
...
+        display: -webkit-box; // stylelint-disable-line value-no-vendor-prefix
+        -webkit-box-orient: vertical;
+        overflow: hidden;
+        align-self: center;
+        white-space: normal;
+        // by default clamp 2 lines
+        -webkit-line-clamp: 2;
+    }
+
+    @each $size in $supportedLineCountSizes {
```
Comment (amje) [inline src/components/Text/text/text.scss:88]: We generate ton of CSS for no reason. Why don't we just let `ellipsisLines` to be a `number` type instead? There is no need in second prop `ellipsisLinesCount`.

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Text/text/text.scss (hunk 4 above)
Comment (IsaevAlexandr) [reply src/components/Text/text/text.scss:88]: it's only for inline text util usage, a remove this

File: src/components/Text/Text.tsx (hunk 2 above)
Comment (IsaevAlexandr) [reply src/components/Text/Text.tsx]: it's not obvious to use max width with this component to work. So now it's build in

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

File: src/components/Text/Text.tsx [hunk 5]
```diff
@@ -48,39 +54,54 @@ type TextPropsWithoutRef<C extends React.ElementType> = {as?: C} & Omit<TextProp
...
-            </Tag>
-        );
-    },
-) as (<C extends React.ElementType = 'span'>({
+    const style: React.CSSProperties = {
+        ...outerStyle,
+    };
+
+    if (typeof ellipsisLines === 'number') {
+        style.WebkitLineClamp = ellipsisLines;
```
Comment (ogonkov) [inline src/components/Text/Text.tsx:81]: What the point to use inline style, and not modifier in stylesheet?

**Pull request merged**